view/next-project/src/pages/purchasereports/index.tsx

```diff
@@ -104,18 +120,24 @@ export default function PurchaseReports(props: Props) {
     }
   }, []);
 
-  // すべてのpurchaseReportの合計金額
+  //すべてのpurchaseReportの合計金額
   const totalReportFee = useMemo(() => {
-    if (filteredPurchaseReportViews) {
+    if (purchaseReportViews) {
       let totalFee = 0;
-      filteredPurchaseReportViews.map((purchaseReportView: PurchaseReportView) => {
-        totalFee +=
+      purchaseReportViews.forEach((purchaseReportView: PurchaseReportView) => {
+        const fee =
           TotalFee(purchaseReportView.purchaseReport, purchaseReportView.purchaseItems) || 0;
+        console.log(`Report ID: ${purchaseReportView.purchaseReport.id}, Fee: ${fee}`);
+        totalFee += fee;
       });
       return totalFee;
     }
     return 0;
-  }, [filteredPurchaseReportViews]);
+  }, [TotalFee, purchaseReportViews]);
+
+  useEffect(() => {
+    getPurchaseReport();
+  }, [selectedYear, getPurchaseReport]);
```

Comment:
```suggestion
  }, [selectedYear]);
```
useEffectで副作用に`getPurchaseReport`を入れると無限ループになっているので修正してください

Reply:
修正しました。確認お願いします！